genesyscloud/resource_genesyscloud_flow.go

```diff
@@ -92,6 +92,16 @@ func ResourceFlow() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
+			"flow_name": {
+				Description: `Genesys Cloud flow name`,
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
```

Comment:
Same here. `Computed: true`

Reply:
I have another concern about this field. If the end user wants to use it to define the name of the flow (which they will assume is possible when they see the field) , they might try to do so even though the flow name in the yaml file is different. That should also cause errors.

One possible workaround: You could specify in the docs that the name has to match the name field in the yaml file, and do some validation to ensure that is the case. There is also the possibility that they are using the substitutions map to populate the name field, in which case you would need to check inside that map, find the key, and again verify the two names are the same.

Reply:
(Same for flow_type)

Reply:
That is do-able since the exported contents is guaranteed to have the correct flow name and type values in the resource... but then the problem is that the exported flow file could be edited by the user to change the name. In that case, the user will have to remember to change the resource flow_name field before applying the export config.

Reply:
@HemanthDogiparthi12 I submitted a PR last week to update the resource name with the flowtype as you mentioned: https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/820

Reply:
Yes. Thanks @bbbco 
Hi @williamjeuniaux Based on the updated resource names , do you think we still need flow name and flow type attributes ?

Reply:
Hi @HemanthDogiparthi12 

PR 820 doesn't solve the name problem.

What special character do you recommend using? From what I can gather, it looks like it may not be best practices and could lead to issues while referencing the resource. I also checked if GC wasn't allowing hyphen "-", but they do, so we can't replace the space by it.

I do understand this might create ambiguity without the proper documentation. I started implementing changes following Charlie's comment about documenting the behavior and doing the necessary validation to make sure the values are the same as the ones in the yaml. If you think of a better way, I'll be happy to change it of course, let me know.

The flow type attribute is a nice to have, since the type is already in the name of the resource. However, from a structural perspective, it appears more logical to incorporate the type as an attribute of the flow object.